tools/autograd/gen_variable_type.py

```diff
@@ -71,12 +71,18 @@
 # arguments (inside of the `native_functions.yaml`)
 RENAME_TRACE_ADD_ARGS = {
     'fill': '''\
-    jit::tracer::addInputs(node, "options", TensorOptions());
+    jit::tracer::addInputs(node, "dtype", c10::optional<ScalarType>());
```

Comment:
good point, I should change it to the default instead. Not sure what's the right thing to do but I want to avoid behavior changes for now.

Reply:
ugh turns out this actually used `TensorOptions::dtype_opt()`, so it would in fact trace a `nullopt`. But `layout, device, pin_memory` used default values. This is almost certainly wrong but I'm replicating the logic for now.